src/mon/PGMonitor.cc

```diff
@@ -1310,9 +1310,19 @@ void PGMonitor::_try_mark_pg_stale(
 
 bool PGMonitor::check_down_pgs()
 {
-  dout(10) << "check_down_pgs" << dendl;
+  dout(10) << "check_down_pgs last_osdmap_epoch "
+	   << pg_map.last_osdmap_epoch << dendl;
+  if (pg_map.last_osdmap_epoch == 0)
+    return false;
+
+  // use the OSDMap that matches the one pg_map has consumed.
+  std::unique_ptr<OSDMap> osdmap;
+  bufferlist bl;
+  int err = mon->osdmon()->get_version_full(pg_map.last_osdmap_epoch, bl);
```

Comment:
the OSDMonitor kicks pgmon in `OSDMonitor::update_from_paxos()`, so the pgmon can consume (`check_osd_map()`) the latest osdmaps once the monitor is **active**[1]. and pg_map.last_osdmap_epoch is updated in `PGMonitor::update_from_paxo()` when the monitor is still **updating**.

so, is there a chance that `check_down_pgs()` gets called **after** `pg_map.last_osdmap_epoch` is updated, and **before** `check_osd_map()` is called? in other words:

1. pg_map.set_last_osdmap_epoch(last_osdmap_epoch);
1. PGMonitor::check_down_pgs()
1. check_osd_map()

---
[1] see `PGMonitor::check_osd_map()`
```c++
  if (!is_writeable()) {
    dout(10) << __func__ << " -- pgmap not writeable, waiting" << dendl;
    wait_for_writeable_ctx(new RetryCheckOSDMap(this, epoch));
    return;
  }
```

the monitor is not writable until it is done with refresh (the `update_from_paxos()` calls).